Use linux-musl runtime on Alpine Helix workers

nattress opened the pull request:
Fixes https://github.com/dotnet/runtime/issues/36825

nattress committed:
Use linux-musl runtime on Alpine Helix workers

jkoritzinsky commented:
Link to an outerloop build: https://dev.azure.com/dnceng/public/_build/results?buildId=655809&view=results

jkoritzinsky commented:
Formatting job failed to download clang-format

trylek reviewed (approved):
Looks just as beautiful as Jeremy's version, thanks for fixing!

jkoritzinsky reviewed (changes requested):
Outerloop job is still failing with the same issue. Apparently TargetOS is getting passed in as Linux when we send the tests to Helix.

jkoritzinsky commented:
We need to change the following line to also account for the `osSubGroup` parameter (which I believe will also need to be added to the parameter list at the top of the YAML file): https://github.com/dotnet/runtime/blob/a2ec75f3feaabec604fb03ad5252c92d3e152aa9/eng/pipelines/common/templates/runtimes/send-to-helix-step.yml#L42

nattress committed:
Add OS Subgroup to send-to-helix-step

nattress commented:
Do we need to update the thing that invokes this YAML to also pass in `osSubGroup` or will it just work?

trylek commented:
I'm afraid that the former is true, there's nothing about YAML that just works...

trylek commented:
https://github.com/dotnet/runtime/blob/a2ec75f3feaabec604fb03ad5252c92d3e152aa9/eng/pipelines/common/templates/runtimes/run-test-job.yml#L283

nattress committed:
Pass in osSubgroup

Also, capitalize it in line with the rest of the repo

trylek commented on eng/pipelines/common/templates/runtimes/send-to-helix-step.yml:
```diff
@@ -82,7 +82,7 @@ steps:
     env:
       __BuildArch: ${{ parameters.archType }}
-      __TargetOS: ${{ parameters.osGroup }}${{ parameters.osSubGroup }}
+      __TargetOS: ${{ parameters.osGroup }}${{ parameters.osSubgroup }}
```
Looks good. At some point we may want to consider splitting these too into something like __TargetOSGroup / Subgroup but please avoid doing that as part of this change :-).

nattress commented:
Thanks Tomas 😄 Btw, when I fixed my osSubgroup casing, I grepped the repo and found one other place we capitalize it differently: https://github.com/dotnet/runtime/blob/bdc09433b6d451e66943de959afdab92117c502e/eng/pipelines/libraries/base-job.yml#L65

My understanding is YAML is case sensitive so this is probably not working right.

trylek commented:
YAML is mostly case insensitive (not only when resolving variable names, even when comparing strings) so this should be technically fine but I don't deny I find the Pascal casing of SubGroup just as annoying as in case of RunTime. Please feel free to fix that one as part of this change if you have the energy to do so.

nattress committed:
Use __TargetOS which understands linux_musl for rid generation

I don't fully understand why we use `TargetOS` in `helixpublishwitharcade.proj` but set `__TargetOS` in the calling `send-to-helix-step.yml`.
